source/byte_buf.c

```diff
@@ -24,9 +24,8 @@
 #endif
 
 int aws_byte_buf_init(struct aws_byte_buf *buf, struct aws_allocator *allocator, size_t capacity) {
-    if (!buf || !allocator) {
-        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
-    }
+    AWS_ERROR_PRECONDITION(buf);
+    AWS_ERROR_PRECONDITION(allocator);
```

Comment:
These two I think can be not-the-ERROR variant. If you pass `NULL` for `buf` or `allocator` you should probably eat the assert.

Reply:
```FATAL_ASSERT``` or ```assert```?

Reply:
I don't feel strongly either way, but I think vanilla `AWS_PRECONDITION` is more prevalent, maybe stick with that?

Reply:
Done